core/src/musig2.rs

```diff
@@ -125,10 +125,32 @@ mod tests {
         utils,
     };
     use bitcoin::{hashes::Hash, script, Amount, OutPoint, ScriptBuf, TapNodeHash, TxOut, Txid};
-    use secp256k1::{rand::Rng, Keypair, Message};
+    use secp256k1::{rand::Rng, Keypair, Message, PublicKey, XOnlyPublicKey};
 
     use super::{nonce_pair, MuSigNoncePair};
 
+    trait FromPublicKeys {
+        fn from_pks(
+            pks: Vec<PublicKey>,
+            tweak: Option<[u8; 32]>,
+        ) -> (musig2::secp256k1::PublicKey, XOnlyPublicKey);
+    }
+
+    impl FromPublicKeys for XOnlyPublicKey {
+        fn from_pks(
```

Comment:
Do you think we need to describe this from_pks is using musig2?

Reply:
Good point, I renamed it to `from_musig2_pks`. Let me know if you have other ideas for the name.

Although on another look, I think since it is a private trait inside `musig2` module it might be self-explanatory.

Reply:
Good point, I renamed it to `FromMuSig2Pks::from_musig2_pks`. Let me know if you have other ideas for the name.

On another look, I think since it is a private trait inside `musig2` module it might be redundant. On the other hand, it is true that there are other kinds of public keys this could be confused with.